src/seedu/addressbook/data/person/address/Address.java

```diff
@@ -0,0 +1,148 @@
+package seedu.addressbook.data.person.address;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Person's address in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
+ */
+public class Address {
+
+    public static final String EXAMPLE = "123, some street, some unit, 123456";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses should be [Block], [Street], [Unit], [Postal Code]";
+    public static final String ADDRESS_VALIDATION_REGEX = ".+, .+, .+, .+";
+    public static final String ADDRESS_EXTRACT_REGEX = ", ";
+
+    private static final int ADDRESS_BLOCK_INDEX = 0;
+    private static final int ADDRESS_STREET_INDEX = 1;
+    private static final int ADDRESS_UNIT_INDEX = 2;
+    private static final int ADDRESS_POSTAL_CODE_INDEX = 3;
+
+    public final String address;
+
+    private Block block;
+    private Street street;
+    private Unit unit;
+    private PostalCode postalCode;
+    
+    private boolean isPrivate;
+
+    /**
+     * Validates given address.
+     *
+     * @throws IllegalValueException if given address string is invalid.
+     */
+    public Address(String address, boolean isPrivate) throws IllegalValueException {
+        String trimmedAddress = address.trim();
+        this.isPrivate = isPrivate;
+        if (!isValidAddress(trimmedAddress)) {
+            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        }
+        this.address = trimmedAddress;
```

Comment:
Why are you still retaining the actual value of the address? Consider the havoc duplication can create!